content/tutorials/pubnub-adaptor.textile

```diff
@@ -0,0 +1,199 @@
+
+h4. Connections
+
+* In Pubnub's model, message continuity is available for '5-20 minutes' and up to 100 messages. To provide compatibility with this, the Pubnub adaptor will stay connected and attached to your channels for 5 minutes after the last subscribe long poll it receives. After 5 minutes, the adaptor will disconnect from Ably, and any subsequent subscribe poll will be treated as a new connection (and will get messages from that point in time onwards).
+* During development, using a lot of different uuids (eg using the @unique_uuid@ option of the Pubnub js lib to generate a new uuid on each page refresh) may result in connections racking up quickly, as each will stay alive for at least 5 minutes. We recommend avoiding that option, at least during development.
+
+h2. Download tutorial source code
+
+blang[javascript].
+  The complete source code of a working example of Ably and Pubnub client libraries communicating with each other "is available on Github":https://github.com/ably/tutorials/commits/pubnub-adaptor-javascript.
+
+  We recommend that you clone the repo locally:
+
+  bc[sh]. git clone https://github.com/ably/tutorials.git
+
+  Checkout the tutorial branch:
+
+  bc[sh]. git checkout pubnub-adaptor-javascript
+
+  And then run the demo locally by adding your "Ably API key":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys to @example.html@ and opening the page in your browser.
+
+blang[nodejs].
+  The complete source code of a working example of Ably and Pubnub client libraries communicating with each other "is available on Github":https://github.com/ably/tutorials/commits/pubnub-adaptor-nodejs.
+
+  We recommend that you clone the repo locally:
+
+  bc[sh]. git clone https://github.com/ably/tutorials.git
+
+  Checkout the tutorial branch:
+
+  bc[sh]. git checkout pubnub-adaptor-nodejs
+
+  And then run the demo locally by adding your "Ably API key":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys to @example.js@ and running the demo @node example.js@
+
+blang[ruby].
+  The complete source code of a working example of Ably and Pubnub client libraries communicating with each other "is available on Github":https://github.com/ably/tutorials/commits/pubnub-adaptor-ruby.
+
+  We recommend that you clone the repo locally:
+
+  bc[sh]. git clone https://github.com/ably/tutorials.git
+
+  Checkout the tutorial branch:
+
+  bc[sh]. git checkout pubnub-adaptor-ruby
+
+  And then run the demo locally by adding your "Ably API key":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys to @example.rb@ and running the demo with @bundle exec ruby example.rb@
+
+h2. Next steps
+
+# Full usage notes for the Pubnub adaptor are available at: "Using the Ably Pubnub Protocol Adaptor":https://support.ably.io/support/solutions/articles/3000055086-using-the-ably-pusher-protocol-adaptor/.
```

Comment:
Can we add the first bullet "[Review the many reasons to migrate from PubNub to Ably](https://www.ably.io/compare/ably-vs-pubnub)"